app/Jobs/DeleteWikiDispatcherJob.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace App\Jobs;
+use App\Wiki;
+use Carbon\Carbon;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Facades\Bus;
+use Throwable;
+use App\WikiSetting;
+use App\QueryserviceNamespace;
+use App\Http\Controllers\WikiController;
+use App\WikiDb;
+use Illuminate\Support\Facades\Config;
+
+class DeleteWikiDispatcherJob extends Job
+{    
+    /**
+     * @return void
+     */
+    public function handle()
+    {
+        $deleteCutoff = Carbon::now()->subDays(
+            Config::get('wbstack.wiki_hard_delete_threshold')
+        );
+        $wikis = Wiki::withTrashed()->whereDate( 'deleted_at', '<=', $deleteCutoff )->get();
+
+        if( !$wikis->count() ) {
+            Log::info( __METHOD__ . ": Found no soft deleted wikis. exiting.");
+            return;
+        }
+
+        foreach($wikis as $wiki) {
+            Log::info( __METHOD__ . ": Starting hard-delete chain for id: {$wiki->id}, domain: {$wiki->domain}.");
+
+            $jobs = [];
+
+            $namespace = QueryserviceNamespace::whereWikiId($wiki->id)->first();
+
+            // if there is a namespace add the job for that
+            if( $namespace ) {
+                $jobs[] = new DeleteQueryserviceNamespaceJob( $namespace->id );
```

Comment:
This is passing the namespace->, but the job actually takes a wiki id here!

Reply:
geez

Reply:
good spot, thanks

Reply:
This would be one reason to argue FOR typing / passing an object here, and against just ints